x-pack/plugins/infra/server/saved_objects/inventory_view/types.ts

```diff
@@ -5,20 +5,14 @@
  * 2.0.
  */
 
-import { isoToEpochRt, nonEmptyStringRt } from '@kbn/io-ts-utils';
+import { isoToEpochRt } from '@kbn/io-ts-utils';
 import * as rt from 'io-ts';
-
-export const inventoryViewSavedObjectAttributesRT = rt.intersection([
-  rt.strict({
-    name: nonEmptyStringRt,
-  }),
-  rt.UnknownRecord,
-]);
+import { inventoryViewAttributesRT } from '../../../common/inventory_views';
 
 export const inventoryViewSavedObjectRT = rt.intersection([
   rt.type({
     id: rt.string,
-    attributes: inventoryViewSavedObjectAttributesRT,
+    attributes: inventoryViewAttributesRT,
```

Comment:
**note:** I think coupling the attributes to the type defined and used also for the API responses revert the changes made for the migrations to [versioned APIs](https://docs.elastic.dev/kibana-dev-docs/versioning-http-apis#1-do-not-directly-expose-persistence-schemas-on-your-http-api-endpoints), where is states the persistence schema shouldn't be directly exposed. In terms of SO attributes definition, I think we should approach it as done for the logView SO on [this PR](https://github.com/elastic/kibana/pull/158098). Wdyt?

Reply:
Good catch! thanks